### PR TITLE
[SPARK-23310][CORE] Turn off read ahead input stream for unshafe shuffle reader

### PR DESCRIPTION
**Pull request by sitalkedia**
To fix regression for TPC-DS queries

### CODE REVIEW

File: core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public UnsafeSorterSpillReader(
              SparkEnv.get().conf().getDouble("spark.unsafe.sorter.spill.read.ahead.fraction", 0.5);
 
     final boolean readAheadEnabled = SparkEnv.get() != null &&
-        SparkEnv.get().conf().getBoolean("spark.unsafe.sorter.spill.read.ahead.enabled", true);
+        SparkEnv.get().conf().getBoolean("spark.unsafe.sorter.spill.read.ahead.enabled", false);
```
Comment (sameeragarwal) [inline core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java]: Can we add a comment here to add more context? Perhaps also link a JIRA/TODO to re-enable this in 2.4?

**Comment by sameeragarwal** [comment]
Just a minor comment, LGTM. Thanks!

File: core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java (hunk 1 above)
Comment (sitalkedia) [reply core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java]: Good point, added more context to it.

**Comment by kiszk** [comment]
LGTM

**Comment by sameeragarwal** [comment]
merging this to master/2.3

**Pull request closed without merging**